brooklyn-server/core/src/main/java/org/apache/brooklyn/core/typereg/AbstractTypePlanTransformer.java

```diff
@@ -103,7 +103,8 @@ public Object create(final RegisteredType type, final RegisteredTypeLoadingConte
                 @Override protected Object visitSpec() {
                     try { 
                         AbstractBrooklynObjectSpec<?, ?> result = createSpec(type, context);
-                        result.catalogItemId(type.getId());
+                        // see notes in CampResolver setting catalogItemId; we might need to keep both
```

Comment:
I do see any new notes in `CampResolver` - did you mean in `AbstractBrooklynObjectSpec`?